### PR TITLE
Use kubeclient with non-blocking IO support

### PR DESCRIPTION
**Pull request by msufa**
Remove the watcher monkey patch once https://github.com/abonas/kubeclient/pull/142 gets merged to kubeclient.

/cc @henders @sbrnunes @jonmoter 

### References
 - Jira link: 

### Risks
 - None

### CODE REVIEW

**Comment by msufa** [comment]
The corresponding PR in `kubeclient` has been merged and tagged to `v1.1`. This PR is no longer WIP and is ready to be reviewed.

File: Gemfile.lock [hunk 1]
```diff
@@ -179,6 +178,7 @@ GEM
       dotenv (= 0.11.1)
     erubis (2.7.0)
     eventmachine (1.0.8)
+    eventmachine (1.0.8-java)
```
Comment (henders) [inline Gemfile.lock]: Don't know why it added this gem! Maybe remove it?

**Comment by henders** [comment]
:+1:

File: Gemfile.lock (hunk 1 above)
Comment (steved) [inline Gemfile.lock]: we need to remove java from the platforms once and for all

**Pull request merged**